flashinfer/green_ctx.py

```diff
@@ -57,6 +74,29 @@ def split_resource(
     return results, remaining
 
 
+def split_resource_by_sm_count(
+    cu_dev: CUdevice, resource: CUdevResource, sm_counts: List[int]
+) -> Tuple[List[CUdevResource], CUdevResource]:
+    results = []
+    for sm_count in sm_counts:
+        result, remaining = split_resource(resource, 1, sm_count)
+        results.extend(result)
+        # Refresh the remaining resource for the next iteration
+        desc = checkCudaErrors(driver.cuDevResourceGenerateDesc([remaining], 1))
+        green_ctx = checkCudaErrors(
+            driver.cuGreenCtxCreate(
+                desc, cu_dev, driver.CUgreenCtxCreate_flags.CU_GREEN_CTX_DEFAULT_STREAM
+            )
+        )
+        resource = checkCudaErrors(
+            driver.cuGreenCtxGetDevResource(
+                green_ctx, driver.CUdevResourceType.CU_DEV_RESOURCE_TYPE_SM
+            )
+        )
+
```

Comment:
Add some logging information about the success green context creation, and the real number of SM for each partition.